Source/Widgets/Viewer/Viewer.js

```diff
@@ -524,14 +580,46 @@ Either specify options.terrainProvider instead or set options.baseLayerPicker to
                     timeline.resize();
                 }
             });
-        } else if (defined(timeline)) {
-            timeline.container.style.right = 0;
+        }
+
+        // VR
+        var vrButton;
+        var vrSubscription;
+        var vrModeSubscription;
+        var vrCallback;
+        if (options.vrButton === true) {
+            var vrContainer = document.createElement('div');
+            vrContainer.className = 'cesium-viewer-vrContainer';
+            viewerContainer.appendChild(vrContainer);
+            vrButton = new VRButton(vrContainer, cesiumWidget.scene, options.fullScreenElement);
+
+            vrSubscription = subscribeAndEvaluate(vrButton.viewModel, 'isVREnabled', function(isVREnabled) {
+                vrContainer.style.display = isVREnabled ? 'block' : 'none';
+                if (defined(fullscreenButton)) {
+                    vrContainer.style.right = fullscreenContainer.clientWidth + 'px';
+                }
+                if (defined(timeline)) {
+                    timeline.container.style.right = vrContainer.clientWidth + 'px';
+                    timeline.resize();
+                }
+            });
+
+            vrModeSubscription = subscribeAndEvaluate(vrButton.viewModel, 'isVRMode', function(isVRMode) {
+                enableVRUI(that, isVRMode);
+            });
+
+            vrCallback = function() {
+                enableVRUI(that, false);
+            };
+            document.addEventListener(Fullscreen.changeEventName, vrCallback);
```

Comment:
Keep in mind that a successful change into fullscreen mode triggers one of these events.  You have it wired to call `enableVRUI` both in response to viewModel changes and in response to the browser event.  So, enabling it in the viewModel causes a brief but successful switch into fullscreen mode, followed by an event and a re-submission of the request for fullscreen.  Is is that second request that is denied by Chrome, generating the warning in the console, and bailing out of fullscreen mode that was enabled by the first.

Also, using the non-VR fullscreen button generates these events too, and inadvertently enables VR mode.